### PR TITLE
Language detection via franc

### PR DESCRIPTION
**Pull request by kwonoj**
This is experimental POC PR for discussion, moving language detention to `franc` from `cld`, as well as removing relying on electron-remote to detect languages.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,7 +1,7 @@
 {
-  "name": "electron-spellchecker",
+  "name": "@kwonoj/electron-spellchecker",
```
Comment (CharlieHess) [inline package.json]: Instead of publishing to npm you could change the dependency declaration in `package.json` to use your fork

Comment (kwonoj) [inline package.json]: thanks for tip :bow:

**Comment by CharlieHess** [comment]
I'm in favor of this, esp if we can prove out `franc`.

Discussed with OJ offline and came up with more suggestions:

* [Whitelist](https://github.com/wooorm/franc#whitelist) `franc` to only the dictionaries Chromium supports
* Use an isomorphic `fetch` to download dictionaries and :knife: the taskpool altogether

**Comment by kwonoj** [comment]
Working on whitelist now.

**Comment by paulcbetts** [comment]
What's the advantage of franc over cld?

**Comment by CharlieHess** [comment]
(Slack tries to detect hint text whenever the channel changes, spinning up a new renderer process each time. It's a bit too much overhead on resource-constrained systems).

**Comment by kwonoj** [comment]
PR updated to include whitelist, and passes all tests as well. @CharlieHess 

image

(ignore one failng test, it's my local machine doesn't have `en` locale kbd layout)

**Comment by paulcbetts** [comment]
> a synchronous API lets us avoid the taskpool.

Sure, but now your language detection is running on the same thread as the one processing input, no? Doesn't this introduce typing lag?

**Comment by CharlieHess** [comment]
@paulcbetts nah, we just dropped it in an idle callback. Spellcheck has to return immediately but AFAICT language detection can complete whenever.

**Comment by kwonoj** [comment]
I'm closing this PR as I'll not going to use franc anymore, instead will try to hook up other way around.

**Pull request closed without merging**